docs/apache-airflow/concepts/scheduler.rst

```diff
@@ -138,12 +141,101 @@ The following databases are fully supported and provide an "optimal" experience:
+optimizations (we will not recommend any specific tools - just use the tools that you usually use
+to observe and monitor your systems):
+
+* decide which aspect of performance is most important for you (what you want to improve)
+* observe your system to see where your bottlenecks are: CPU, memory, I/O are the usual limiting factors
+* based on your expectations and observations - decide what is your next improvement and go back to
+  the observation of your performance, bottlenecks. Performance improvement is an iterative process
+
+The improvements that you can consider are:
+
+* improve utilization of your resources. This is when you have a free capacity in your system that
+  seems underutilized (again CPU, memory I/O, networking are the prime candidates) - you can take
+  actions like increasing number of schedulers, parsing processes or decreasing intervals for more
+  frequent actions might bring improvements in performance at the expense of higher utilization of those.
+* increase hardware capacity (for example if you see that CPU is limiting you or tha I/O you use for
+  DAG filesystem is at its limits). Often the problem with scheduler performance is
+  simply because your system is not "capable" enough and this might be the only way. For example if
+  you see that you are using all CPU you have on machine, you might want to add another scheduler on
+  a new machine - in most cases, when you add 2nd or 3rd scheduler, the capacity of scheduling grows
+  linearly (unless the shared database or filesystem is a bottleneck).
+* experiment with different values for the "scheduler tunables". Often you might get better effects by
+  simply exchanging one performance aspect for another. For example if you want to decrease the
+  cpu usage, you might increase file processing interval (but the result will be that new DAGs will
+  appear with bigger delay). Usually performance tuning is the art of balancing different aspects.
+* sometimes you change scheduler behaviour slightly (for example change parsing sort order)
+  in order to get better fine-tuned results for your particular deployment.
+
+In order to perform fine-tuning, it's good to understand how Scheduler works under-the-hood.
+You can take a look at the ``Airflow Summit 2021``
+`Deep Dive into the Airflow Scheduler talk <https://youtu.be/DYC4-xElccE>`_ to perform the fine-tuning.
+
+Here are the most important tunables you can use to impact various performance aspects of the Scheduler:
+
 .. _scheduler:ha:tunables:
 
-Scheduler Tuneables
-"""""""""""""""""""
+Scheduler Tunables
+""""""""""""""""""
 
-The following config settings can be used to control aspects of the Scheduler HA loop.
+The following config settings can be used to control aspects of the Scheduler.
+However you can also look at other scheduler configuration parameters available at
+:doc:`../configurations-ref` in ``[scheduler]`` section.
```

Comment:
Would leave this out, or mention something like "non-performance related configuration options"